pinot-core/src/main/java/com/linkedin/pinot/core/startree/DimensionBuffer.java

```diff
@@ -66,8 +61,8 @@ public boolean equals(Object obj) {
   public String toString() {
     StringBuilder builder = new StringBuilder("[");
     String delimiter = "";
-    for (int i = 0; i < _numDimensions; i++) {
-      builder.append(delimiter).append(_intBuffer.get(i));
+    while (_buffer.remaining() > 0) {
```

Comment:
Seems like stateful toString(), which is incorrect. Please fix.

Reply:
Good catch